pkg/service/model_build_target_group.go

```diff
@@ -113,6 +113,10 @@ func (t *defaultModelBuildTask) buildTargetGroupHealthCheckConfigDefault(ctx con
 	if err != nil {
 		return nil, err
 	}
+	healthCheckTimeoutSeconds, err := t.buildTargetGroupHealthCheckTimeoutSeconds(ctx, t.defaultHealthCheckTimeout)
```

Comment:
build the timeout based on feature gate. For example, create a new feature gate `NLBHealthCheckTimeout` set to true by default. Refer to the pkg/config/feature_gates.go for more details. 
You can check it as t.featureGates.Enabled(config.NLBHealthCheckTimeout)

Reply:
I named it `ServiceHealthCheckTimeout` to be more consistent with naming.

Reply:
Health check configuration for ALB target groups can also be specified via service annotations, that was the reason I wanted to specifically say NLBHealthCheckTimeout

Reply:
> Health check configuration for ALB target groups can also be specified via service annotations, that was the reason I wanted to specifically say NLBHealthCheckTimeout

But the annotation is different: ingress and service: 
`service.beta.kubernetes.io/aws-load-balancer-healthcheck-timeout`
vs 
`alb.ingress.kubernetes.io/healthcheck-timeout-seconds`

Reply:
You're right, the annotations are different. However, in case of ALB, you could also specify the some of the `alb.ingress.kubernetes.io/*` annotations on a backend service which would affect the particular target group.